Libraries/Database.psm1

```diff
@@ -74,6 +74,96 @@ Function Get-SQLQueryOfTelemetryData ($TestPlatform, $TestLocation, $TestCategor
+					if ($azAccountType -eq "ServicePrincipal") {
+						$applicationId = $global:XmlSecrets.secrets.SubscriptionServicePrincipalClientID
+						$authority = $global:XmlSecrets.secrets.SubscriptionServicePrincipalTenantID
+						$applicationKey = $global:XmlSecrets.secrets.SubscriptionServicePrincipalKey
+						$kcsb = $kcsb.WithAadApplicationKeyAuthentication($applicationId, $applicationKey, $authority)
+					}
+					elseif ($azAccountType -eq "ManagedService") {
+						$msiClientId = $global:XmlSecrets.secrets.MsiClientId
+						$kcsb = $kcsb.WithAadManagedIdentity($msiClientId)
+					}
+					$queryProvider = [Kusto.Data.Net.Client.KustoClientFactory]::CreateCslQueryProvider($kcsb)
+					# Query first to get table schema, and prepare ingest inline command
+					$crp = New-Object Kusto.Data.Common.ClientRequestProperties
+					$crp.ClientRequestId = "MyPowershellScript.ExecuteQuery." + [Guid]::NewGuid().ToString()
+					$crp.SetOption([Kusto.Data.Common.ClientRequestProperties]::OptionServerTimeout, [TimeSpan]::FromSeconds(30))
+					# Here may throw exception, when $tableName does not exist
+					$queryReader = $queryProvider.ExecuteQuery("$tableName | limit 1", $crp)
+					$queryDataTable = [Kusto.Cloud.Platform.Data.ExtendedDataReader]::ToDataSet($queryReader).Tables[0]
+					$kustoTableColumns = @($queryDataTable.Columns.ColumnName)
+					foreach ($rowValue in $valueArr) {
+						$ingestValueArray = @($kustoTableColumns)
+						$cellValueIndex = 0
+						$cellValueArray = $rowValue.Split(",").Trim("' ")
+						for ($tableColumnIndex = 0; $tableColumnIndex -lt $kustoTableColumns.Count; $tableColumnIndex++) {
+							if ($columns -contains $kustoTableColumns[$tableColumnIndex]) {
+								$ingestValueArray[$tableColumnIndex] = $cellValueArray[$cellValueIndex++].Replace(",", " ")
+							}
+							else {
+								$ingestValueArray[$tableColumnIndex] = ""
+							}
+						}
+						$ingestInlineCmd += " [ $($ingestValueArray -Join ',') ]"
+					}
+					# Execute ingest command with AdminProvider
+					$adminProvider = [Kusto.Data.Net.Client.KustoClientFactory]::CreateCslAdminProvider($kcsb)
+					# Here may throw exception, when schema does not match
+					$adminReader = $adminProvider.ExecuteControlCommand($ingestInlineCmd)
+					if ($adminReader.HasRows -and $adminReader["ExtentId"] -ne [Guid]::Empty) {
+						Write-LogInfo "Ingested successfully into '$kustoDatabase' of '$kustoClusterURI' with `n $ingestInlineCmd "
+					}
+				}
+				else {
+					Write-LogErr "Missing 'tablename' or 'columns' or 'values' from SQLString '$SQLString', hence skip Kusto ingestion"
+				}
+			}
+			else {
+				Write-LogErr "Failed to parse SQLString '$SQLString', hence skip Kusto ingestion"
+			}
+		}
+	}
+	catch {
+		Write-LogErr "Ingest Kusto Data from TSQLString:  ERROR"
+		$line = $_.InvocationInfo.ScriptLineNumber
+		$script_name = ($_.InvocationInfo.ScriptName).Replace($PWD, ".")
```

Comment:
`$script_name` this variable naming is not following pattern like "variableName". Is there any particular reason?

Reply:
no special reason, just (copied) code from otherwhere.